### PR TITLE
Room controller refactoring

### PR DESCRIPTION
**Pull request by dvinciguerra**
### Description
This PR don't change anything... its just a refactor to make codebase simpler removing actions logic from routes file.

### How to test?
All things need to continue the same behaviour, principally room actions.

### Expected behavior
N/A

_Based on [expressjs route-separation](https://github.com/expressjs/express/blob/master/examples/route-separation)_

### CODE REVIEW

File: backend/app/controllers/rooms.controller.js [hunk 1]
```diff
@@ -65,4 +65,40 @@ const fetchRooms = (strategy) => {
...
+    temporary: true,
+  };
+
+  const found = req.app.locals.roomsDetail.find(
+    element => element.id == req.query.roomId,
+  );
+
+  if (!found) {
+    req.app.locals.roomsDetail.splice(1, 0, newRoom);
+  }
```
Comment (alesshh) [inline backend/app/controllers/rooms.controller.js]: I think that is a great time to move rooms creation / fetch logic to a proper service or store interface.

That will allow us to create different rooms store strategies, like a backoffice endpoint.
What you think?

File: backend/app/controllers/rooms.controller.js [hunk 2]
```diff
@@ -65,4 +65,40 @@ const fetchRooms = (strategy) => {
...
+
+const list = (req, res) => {
+  res.json(req.app.locals.roomsDetail);
+};
+
+export default {
+  list,
+  create,
+  remove,
+  fetchRooms,
```
Comment (alesshh) [inline backend/app/controllers/rooms.controller.js]:
```suggestion
  fetchAll,
```

Comment (alesshh) [reply backend/app/controllers/rooms.controller.js]: Please, rename this function.

File: backend/app/app.server.js [hunk 3]
```diff
@@ -70,7 +70,7 @@ app.use((req, res, next) => {
 
 app.use(routes);
 
-fetchRooms(ROOMS_SOURCE)
+Room.fetchRooms(ROOMS_SOURCE)
```
Comment (alesshh) [inline backend/app/app.server.js]:
```suggestion
Room.fetchAll(ROOMS_SOURCE)
```

File: backend/app/controllers/rooms.controller.js (hunk 1 above)
Comment (dvinciguerra) [reply backend/app/controllers/rooms.controller.js]: I totally agree!

That's inclusive my next PR that is ready waiting by this context "organization". =)

File: backend/app/controllers/rooms.controller.js (hunk 2 above)
Comment (dvinciguerra) [reply backend/app/controllers/rooms.controller.js]: Done man! =)

**Pull request closed without merging**